src/code/z_actor.c

```diff
@@ -2356,10 +2374,12 @@ void Actor_UpdateAll(PlayState* play, ActorContext* actorCtx) {
 
     player = GET_PLAYER(play);
 
+#if DEBUG_FEATURES
     if (0) {
         // This ASSERT is optimized out but it exists due to its presence in rodata
         ASSERT(gMaxActorId == ACTOR_ID_MAX, "MaxProfile == ACTOR_DLF_MAX", "../z_actor.c", UNK_LINE);
```

Comment:
This is obviously not your change, but perhaps the comment is better said as 
"This assert is optimized out, but it can be assumed to exist because of the string which is present in rodata"